File: hao-ran-music-frontend/src/types/playlist.ts
export type { PlaylistInfo } from '@/api/playlist';
import type { ExactCount } from '@/utils/exactCount';
export interface PlaylistSong {
    id: string | number;
    title?: string;
    name?: string;
    artist?: string;
    artistNames?: string;
    album?: string;
    albumName?: string;
    duration: number;
    url?: string;
    cover?: string;
    coverUrl?: string;
}
export interface PlaylistDetail {
    id: string | number;
    name: string;
    description?: string;
    cover?: string;
    songCount: number;
    playCount: ExactCount;
    isPublic: boolean | number;
    creatorName: string;
    creatorAvatar: string;
    songs: PlaylistSong[];
    allowDownload?: boolean | number;
    allowComment?: boolean | number;
    allowShare?: boolean | number;
}
export interface PlaylistFormData {
    name: string;
    description?: string;
    intro?: string;
    cover?: string;
    isPublic?: boolean | number;
    category?: string;
    language?: string;
    priority?: number;
    tags?: string[];
    playCount?: ExactCount;
    favoriteCount?: ExactCount;
    enablePaid?: boolean;
    monthlyPrice?: number;
}

File: hao-ran-music-backend/src/main/java/com/haoran/music/entity/Playlist.java
package com.haoran.music.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.io.Serializable;
import java.math.BigDecimal;





@Data
@EqualsAndHashCode(callSuper = true)
@TableName("playlist")
public class Playlist extends BaseEntity implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    private String name;
    private String cover;
    private String description;
    private Long userId;
    private Integer type;
    private Integer isPublic;




    private Long songCount;




    private Long playCount;




    private Long favoriteCount;




    private Long visitCount;

    private Integer status;




    private Integer isFeatured;

    private Integer deleted;
    private String tags;
    private String language;
    private Integer isPaid;
    private BigDecimal price;
    private Integer subscribePeriod;






    private Integer allowDownload;




    private Integer allowComment;




    private Integer allowShare;






    private String intro;




    private String category;




    private Long creatorId;
}
